src/vmm/src/vstate/memory.rs

```diff
@@ -540,16 +512,6 @@ mod tests {
         // Write and read from the end of the region
         write_mem(region_last_byte, 0x69);
         assert_eq!(read_mem(region_last_byte), 0x69);
-
-        // Try a read/write operation against the left guard border of the range
```

Comment:
Is that a temporary measure? Do we plan to replace these tests with something else?

Reply:
I was trying to come up with other way to test guard pages, but there are not a lot of options here.
There is an easy way to do it with threads: basically make another thread panic when it receives SIGSEGV and then main thread can catch that panic and verify the cause. But this approach does not work with `musl` for some reason. (works with gnu)

Reply:
I don't think we can differentiate tests base on the gnu/musl compilation.

Reply:
Actually found that `target_env` can be one of 
```
    ""
    "gnu"
    "msvc"
    "musl"
    "sgx"
```
But than we will run unit tests twice. (It can actually be a good thing)

Reply:
Ok, I have returned the guard pages tests and marked them as gnu only tests. I used the panicing thread approach to make it work.